mitodl/ocw-studio
# Ensure that delete_unpublished_courses runs only after all courses have been imported

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #540 

#### What's this PR do?
Uses `celery.group`, `celery.chord`, and `celery.chain` to ensure that the `delete_unpublished_courses` task runs only after all `import_ocw2hugo_course_paths` tasks have completed.

#### How should this be manually tested?
- On master, temporarily adjust the following line in `ocw_import.tasks.import_ocw2hugo_courses` so that the `paths` argument for `delete_unpublished_courses.si` is `course_paths[10:]`:
```python
    if delete_unpublished:
        delete_unpublished_courses_task = delete_unpublished_courses.si(
            paths=course_paths[10:]
        )
```        
- Run the following and you should see some ForeignKeyViolations in the log: `docker-compose run web python manage.py import_ocw_course_sites --bucket ocw-to-hugo-output-qa --limit 30 --chunks 4`

Switch to this branch and again make the same change as above.  Then run the same management command.  There should be no foreign key violations and the same courses should be imported/deleted.
